### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by EvsenevDev**
- [Refactored] Removed useless Objective-C files
- [Refactored] Project Structure for easier navigation
- [Fix] All Warnings

### CODE REVIEW

File: SmartReceipts/Common/UI/AppTheme.swift [hunk 1]
```diff
@@ -44,10 +44,6 @@ class AppTheme: NSObject {
         
         // changes color of active elements like buttons and segments, unfortunately alertview cannot be costumized
         UIApplication.shared.keyWindow?.tintColor = primaryColor
-        
-        // chanes status bar color, however iOS built-in controllers (for example mail composer) may override it
-        UIApplication.shared.isStatusBarHidden = false
```
Comment (wbaumann) [inline SmartReceipts/Common/UI/AppTheme.swift]: How come we got rid of these? A warning in Xcode right?

File: SmartReceipts/Modules/Trips/TripsInteractor.swift [hunk 2]
```diff
@@ -18,7 +18,7 @@ class TripsInteractor: Interactor {
     
     func configureSubscribers() {
         presenter.tripDeleteSubject.subscribe(onNext: { trip in
-            Logger.debug("Delete Trip: \(trip.name)")
+            Logger.debug("Delete Trip: \(trip.name!)")
```
Comment (wbaumann) [inline SmartReceipts/Modules/Trips/TripsInteractor.swift]: For a future change, but can we make `name` non-optional? Then we can remove the optional unwinding in these logs (and elsewhere)

File: SmartReceipts/Common/UI/AppTheme.swift (hunk 1 above)
Comment (EvsenevDev) [reply SmartReceipts/Common/UI/AppTheme.swift]: Yes, this is deprecated property

File: SmartReceipts/Modules/Trips/TripsInteractor.swift (hunk 2 above)
Comment (EvsenevDev) [reply SmartReceipts/Modules/Trips/TripsInteractor.swift]: @wbaumann yes, we can. But this is Objective-C property by default all looks like optional in Swift. I think we avoid it after rewrite from Objective-C to Swift. And we can obviously mark each property as `nonnull` in Objective-C header.

Comment (wbaumann) [reply SmartReceipts/Modules/Trips/TripsInteractor.swift]: Ah makes sense. I forgot that was an Obj-C class 👍

**Pull request merged**